gateway-messages/src/sp_impl.rs

```diff
@@ -786,6 +795,19 @@ fn handle_mgs_request<H: SpHandler>(
         MgsRequest::SetIpccKeyLookupValue { key } => handler
             .set_ipcc_key_lookup_value(sender, port, key, trailing_data)
             .map(|()| SpResponse::SetIpccKeyLookupValueAck),
+        MgsRequest::ReadCaboose { key } => {
+            handler.get_caboose_value(key).map(|data| {
+                if data.len() > crate::MIN_TRAILING_DATA_LEN {
```

Comment:
This is overly conservative - the actual amount of space available for trailing data varies based on the size of the serialized header+message payload, and in the `CabooseValue`case that's quite small; `MIN_TRAILING_DATA_LEN` is the length left after the _largest_ possible message payload. I'm fine leaving this as-is for now; if we end up with caboose values with lengths in between this conservative restriction and what would actually fit (but still in a single packet) we could revisit it, but at that point we probably want to consider multiple packets anyway.